src/gui/widgets/multi_page.hpp

```diff
@@ -134,9 +135,9 @@ class multi_page : public container_base
 
 	/***** ***** ***** setters / getters for members ***** ****** *****/
 
-	void set_page_builder(builder_grid_ptr page_builder)
+	void set_page_builders(const std::map<std::string, builder_grid_const_ptr>& page_builders)
 	{
-		page_builder_ = page_builder;
+		page_builders_ = page_builders;
```

Comment:
I'd recommend ensuring the map is not empty here, in case it's called from somewhere other than the builder.

Either that, or check every time you access `page_builders_.begin()->second` to ensure that `begin()` actually exists.